lib/Backends/BackendUtils.cpp

```diff
@@ -55,3 +55,105 @@ size_t glow::getValueOffset(
   assert(it != symbolTable.end() && "Symbol not found.");
   return it->second.offset;
 }
+
+runtime::RuntimeBundle
+glow::generateRuntimeBundle(const IRFunction *F,
+                            MemoryAllocator *constantAllocator,
+                            MemoryAllocator *placeholderAllocator,
+                            MemoryAllocator *activationsAllocator) {
+  // Handle Constants, Placeholders, and Activations, in that order.
+  /// Symbol table mapping symbol name to offset for runtime.
+  std::unordered_map<std::string, runtime::RuntimeSymbolInfo> symbolTable;
+  uint64_t constantMaxSize = 0;
+  uint64_t placeholderMaxSize = 0;
+  uint64_t activationsMaxSize = 0;
+
+  // Compute the offsets for Constants.
+  if (constantAllocator) {
+    for (auto &v : F->getGraph()->getParent()->getConstants()) {
+      assert(isa<WeightVar>(F->getWeightForNode(v)) && "Expected WeightVar");
+      auto *w = cast<WeightVar>(F->getWeightForNode(v));
+      auto numBytes = w->getSizeInBytes();
+      size_t addr = constantAllocator->allocate(numBytes, v);
+      runtime::RuntimeSymbolInfo symbol;
+      symbol.size = numBytes;
+      symbol.offset = addr;
+      symbol.type = *w->getType();
+      symbolTable.emplace(std::string(v->getName()), symbol);
+    }
+    constantMaxSize = constantAllocator->getMaxMemoryUsage();
```

Comment:
We should only allocate the constants that are used by a given function, not all the constants of the module.

Reply:
I agree I created #2040 and it would also need to address #1599. This PR is just to unify the existing functionality.